From: Igor Mammedov <imammedo@redhat.com>
Date: Wed, 17 Aug 2011 11:53:36 +0200
Subject: [RHEL6.2 Xen PATCH] fix broken build if xen is disabled in config

BZ#722928
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=3562561
RHEL only patch.

Kernel build fails if disable xen is disabled in config, so add 
appropate ifdefs to allow kernel build without xen enabled.

Signed-off-by: Igor Mammedov <imammedo@redhat.com>
---
 arch/x86/kernel/cpu/common.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index 8c8fb23..e866378 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -38,7 +38,9 @@
 #include <asm/uv/uv.h>
 #endif
 
+#ifdef CONFIG_XEN
 #include <xen/xen.h>
+#endif
 
 #include "cpu.h"
 
@@ -284,6 +286,7 @@ struct cpuid_dependent_feature {
 	u32 level;
 };
 
+#ifdef CONFIG_XEN
 static const u32
 xen_dangerous_cpuid_features[] = {
 	/* Mask out GBPAGES & RDTSCP for Xen BZ#703055 */
@@ -314,6 +317,7 @@ static void __cpuinit fltr_xen_cpuid_features(struct cpuinfo_x86 *c, bool warn)
 				x86_cap_flags[*df]);
 	}
 }
+#endif /* CONFIG_XEN */
 
 static const struct cpuid_dependent_feature __cpuinitconst
 cpuid_dependent_features[] = {
@@ -352,12 +356,14 @@ static void __cpuinit filter_cpuid_features(struct cpuinfo_x86 *c, bool warn)
 				x86_cap_flags[df->feature], df->level);
 	}
 
+#ifdef CONFIG_XEN
 	/*
 	 * RHEL Xen HVM guests must filter out additional not masked
 	 * by old kernel-xen features, to avoid crashes.
 	 */
 	if (xen_cpuid_base() != 0)
 		fltr_xen_cpuid_features(c, warn);
+#endif /* CONFIG_XEN */
 }
 
 /*
-- 
1.7.5

